### ISSUE TITLE
Unbound Local in URL Fetcher

### ISSUE THREAD

**Issue by mfeit-internet2** [perfsonar/pscheduler#459]
@szymontrocha observed the following in an email:

I have the following error when trying to run ipv6 test with pscheduler
while bwctl works. Any ideas?
```
[psudouser@psmall-poz1 ~]$ pscheduler task throughput --source
2001:798:fc00:28::6 --destination 2001:808:2:3004:2::2
Submitting task...
Task URL:
https://2001:808:2:3004:2::2/pscheduler/tasks/4b053450-e000-460b-8ce2-d69d057716b3
Traceback (most recent call last):
   File "/usr/libexec/pscheduler/commands/task", line 621, in <module>
     json=True)
   File "/usr/lib/python2.6/site-packages/pscheduler/psurl.py", line
137, in url_get
     __raise_urlexception(status, text, request)
UnboundLocalError: local variable 'request' referenced before assignment


[psudouser@psmall-poz1 ~]$ bwctl -c 2001:808:2:3004:2::2 -s
2001:798:fc00:28::6
bwctl: Using tool: iperf3
bwctl: 17 seconds until test results available

SENDER START
Connecting to host 2001:808:2:3004:2::2, port 5788
[ 14] local 2001:798:fc00:28::6 port 45493 connected to
2001:808:2:3004:2::2 port 5788
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[ 14]   0.00-1.00   sec  7.59 MBytes  63.7 Mbits/sec    0    715 KBytes
[ 14]   1.00-2.00   sec  70.0 MBytes   587 Mbits/sec    0   3.96 MBytes
[ 14]   2.00-3.00   sec   119 MBytes   996 Mbits/sec    0   8.14 MBytes
[ 14]   3.00-4.00   sec   120 MBytes  1007 Mbits/sec    0   13.2 MBytes
[ 14]   4.00-5.00   sec   110 MBytes   923 Mbits/sec  545   10.8 MBytes
[ 14]   5.00-6.00   sec   111 MBytes   933 Mbits/sec    0   11.4 MBytes
[ 14]   6.00-7.00   sec   110 MBytes   923 Mbits/sec    0   11.9 MBytes
[ 14]   7.00-8.00   sec   111 MBytes   933 Mbits/sec    0   12.3 MBytes
[ 14]   8.00-9.00   sec   106 MBytes   891 Mbits/sec   16   1.90 MBytes
[ 14]   9.00-10.00  sec  63.8 MBytes   535 Mbits/sec   29   2.17 MBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[ 14]   0.00-10.00  sec   929 MBytes   779 Mbits/sec 590             sender
[ 14]   0.00-10.00  sec   895 MBytes   751 Mbits/sec                  
receiver

iperf Done.

SENDER END
[psudouser@psmall-poz1 ~]$
```

**Comment by igarny** [comment]
@szymontrocha Szymon please share the results again with -v flag in the bwctl command

**Comment by mfeit-internet2** [comment]
There's no need for additional information.  I already know exactly what the problem is and wrote the ticket for accountability.

**Comment by szymontrocha** [comment]
@igarny please note that bwctl test was working. It's pscheduler command which throws error

**Comment by arlake228** [comment]
Just to clarify here there looks like multiple issues:

1. The fact that the 'request' variable is not initialized which masks any error messages. Mark just fixed that.

2. The fact that the command uses --destination instead of --dest.

3. The fact that even if I change to --dest and hand-patch my local system to initialize request, we still get a "Bad Request" error. Seems like we might have more problems with throughput tests and IPv6.

**Comment by mfeit-internet2** [comment]
Re. #2:  Dan put in multiple variants for the destination, but it's the same option:  https://github.com/perfsonar/pscheduler/blob/master/pscheduler-test-throughput/throughput/cli-to-spec#L62

**Comment by mfeit-internet2** [comment]
More from Andy, from Slack:
```
Task URL:
https://2001:400:301:1150::3/pscheduler/tasks/d34444e8-fcb9-48d4-9007-594b0cfb9ea2
Traceback (most recent call last):
  File "/usr/libexec/pscheduler/commands/task", line 621, in <module>
    json=True)
  File "/usr/lib/python2.6/site-packages/pscheduler/psurl.py", line 137, in url_get
    __raise_urlexception(status, text, request)
  File "/usr/lib/python2.6/site-packages/pscheduler/psurl.py", line 79, in __raise_urlexception
    raise URLException(message)
pscheduler.psurl.URLException: Bad Request
```

**Comment by mfeit-internet2** [comment]
This is _probably_ resolved by 585a556.  Reopen if there are problems.
